pgcontents/query.py

```diff
@@ -370,23 +371,74 @@ def rename_file(db, user_id, old_api_path, new_api_path):
             )
         )
 
-    # Overwriting existing files is disallowed.
-    if file_exists(db, user_id, new_api_path):
-        raise FileExists(new_api_path)
-
-    # Only allow writes to extant directories.
+    # Only allow writes to extant directories
     if not _dir_exists(db, user_id, new_dir):
         raise NoSuchDirectory(new_dir)
 
-    db.execute(
-        files.update().where(
-            _file_where(user_id, old_api_path),
-        ).values(
-            name=new_name,
-            created_at=func.now(),
+    # If we're renaming a file
+    if file_exists(db, user_id, old_api_path):
```

Comment:
All the database-level functions right now operate on **either** directories or files, not both.  I'd rather move this check into PostgresContentsManager.rename_file and have it decide which database-level function to call.